datasource/ethereum/src/block_stream.rs

```diff
@@ -651,97 +525,42 @@ where
                 // This means we need to revert this block.
 
                 // First, load the block in order to get the parent hash.
-                Box::new(ctx.load_block(subgraph_ptr.hash, false).and_then(move |block| {
-                    debug!(
-                        ctx.logger,
-                        "Reverting block to get back to main chain";
-                        "block_number" => format!("{}", block.ethereum_block.block.number.unwrap()),
-                        "block_hash" => format!("{}", block.ethereum_block.block.hash.unwrap())
-                    );
-
-                    // Produce pointer to parent block (using parent hash).
-                    let parent_ptr = EthereumBlockPointer::to_parent(&block.ethereum_block);
-
-                    // Revert entity changes from this block, and update subgraph ptr.
-                    future::result(
-                        ctx.subgraph_store
-                            .revert_block_operations(
-                                ctx.subgraph_id.clone(),
-                                subgraph_ptr,
-                                parent_ptr,
+                Box::new(
+                    self.eth_adapter
+                        .load_block(&ctx.logger, subgraph_ptr.hash)
```

Comment:
Curious why expect this to succeed when we already checked that this block has been uncled, do ethereum node reliably maintain uncles?

Reply:
I didn't change this code, the diff is from formatting churn. This relies on the block ingestor always pulling in the blocks within the reorg threshold.

Reply:
It does. I'm not sure Ethereum nodes keep _everything_ about uncles available, like transaction receipts, but the block headers are stored on chain. This is necessary for e.g. applying and verifying uncle rewards.